chat-widget/src/components/draggable/DraggableEventEmitter.tsx

```diff
@@ -0,0 +1,78 @@
+import React, { ReactNode, useCallback, useEffect, useState } from "react";
+import DraggableEvent from "./DraggableEvent";
+import DraggableEventNames from "./DraggableEventNames";
+
+interface DraggableEventEmitterProps {
+    /**
+     * Unique channel name to send/receive draggable events to prevent event collisions
+     */
+    channel: string;
+    /**
+     * React nodes children
+     */
+    children: ReactNode;
+    /**
+     * HTML element ID of the trigger element to send DraggableEvent to update the draggable element position
+     */
+    elementId: string;
+    /**
+     * Target window to post DraggableEvent messages
+     */
+    targetWindow?: Window;
+}
+
+/**
+ * Trigger component which would send DraggableEvent to the receiver to update the draggable component position
+ *
+ * @param props DraggableEventEmitterProps
+ * @returns
+ */
+const DraggableEventEmitter = (props: DraggableEventEmitterProps) => {
+    const [initialized, setInitialized] = useState(false);
+
+    const postMessage = useCallback((data: DraggableEvent) => {
+        const targetWindow = props.targetWindow ?? window;
+        targetWindow.postMessage(data, "*");
+    }, [props.targetWindow]);
+
+    const dragStart = useCallback((event: MouseEvent) => {
+        postMessage({ channel: props.channel, eventName: DraggableEventNames.DragStart });
+
+        let cursor = { x: event.screenX, y: event.screenY }; // Cursor init position
+        const dragging = (event: MouseEvent) => {
+            event.preventDefault();
+
+            const newX = event.screenX;
+            const newY = event.screenY;
+
+            const offset = { x: newX - cursor.x, y: newY - cursor.y }; // Calculate cursor position diff
+            cursor = { ...cursor, x: newX, y: newY }; // Update cursor new position
+
+            postMessage({
+                channel: props.channel,
+                eventName: DraggableEventNames.Dragging,

```

Comment:
Is there a debounce for this? Will it cause too many events?

Reply:
Debounce would add "lags" to the moving element. So it won't work.